x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/lib/find_custom_dashboard.ts

```diff
@@ -22,3 +22,14 @@ export function findCustomDashboard(
     searchFields: ['assetType'] as [keyof InfraCustomDashboard],
   });
 }
+
+export function findCustomDashboardByDashboardId(
```

Comment:
It's very unlikely that users will link too many dashboards, but to stay on the safe side, perhaps we could limit the `perPage` to 1000 in the function above (findCustomDashboard). I would also sort the list to return the latest linked dashboards -[ same as what APM does](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/server/routes/custom_dashboards/get_custom_dashboards.ts)